src/NServiceBus.Core/Persistence/Msmq/SubscriptionStorage/MsmqSubscriptionStorageQueue.cs

```diff
@@ -51,6 +52,7 @@ public void TryReceiveById(string messageId)
         {
             try
             {
+                //todo: add failing test before fixing this
```

Comment:
Turned out this is a bug, this causes unsubscriptions to not be persisted. I'll write some tests and fix this